packages/language-server/src/edit/handlers/resizeWidthHandler.ts

```diff
@@ -0,0 +1,36 @@
+import { CanvasElement, DefaultEditTypes } from "@hylimo/diagram-common";
+import { IncrementalUpdate, ResizeEdit } from "@hylimo/diagram-protocol";
+import { EditHandler } from "./editHandler.js";
+import { roundToPrecision } from "../../util/roundToPrecision.js";
+
+/**
+ * Handler for resize width edits
+ */
+export const resizeWidth: EditHandler<ResizeEdit> = {
+    type: DefaultEditTypes.RESIZE_WIDTH,
+
+    predictActionDiff(layoutedDiagram, lastApplied, newest, elements) {
+        const updates: IncrementalUpdate[] = [];
+        for (const element of elements) {
+            if (CanvasElement.isCanvasElement(element)) {
+                const sizeToPos = element.width / element.x;
+                element.width = newest.width!;
+                element.x = element.width / sizeToPos;
```

Comment:
Historically speaking, there were no incremental updates, instead, the diagram was updated and sent to the client
However, this was bad for performance, so incremental updates were added, however updating the diagram was never removed.
In general, there really is only 1 case where it makes a difference: if a new client joins while a prediction is ongoing, such a client would see the old diagram instead of the current prediction if it is not updated. That being said, this would only apply until the next prediction, so it probably does not matter at all.
So I think not updating the diagram is the better way forward

Reply:
the exception of course is when a rendered diagram is updated to the newest prediction
in this case the update has to be applied to the diagram, as this diagram will be sent to the client

Reply:
changed with 14be3096870d2ff880f26f59254164e3e254ab61